pkg/controller/database/claim.go

```diff
@@ -180,11 +182,11 @@ func ConfigureMyCloudsqlInstance(_ context.Context, cm resource.Claim, cs resour
 		ResourceSpec: runtimev1alpha1.ResourceSpec{
 			ReclaimPolicy: runtimev1alpha1.ReclaimRetain,
 		},
-		CloudsqlInstanceParameters: rs.SpecTemplate.CloudsqlInstanceParameters,
+		ForProvider: rs.SpecTemplate.ForProvider,
 	}
 
 	if my.Spec.EngineVersion != "" {
```

Comment:
TBH, config parameters having effect on scheduling sounds wild to me. Like, if I specify one parameter `A` in my claim my resource gets provisioned with a set of configurations called `B`. If I don't, it could get to be provisioned with a whole different set of configurations called `C`. If the difference between `B` and `C` is like 10 config, I think we get to have a bias we might not have signed up for.

>If only one of the class or the claim sets the value, use that.
    If both set the same value, use that.
    If both set different values, return an error.

This makes sense to me. If resource class doesn't specify config `K`, then it means it gives the freedom of choice to the claimer, which is fairly intuitive. In the case that neither of them supplied the config and it's optional, then the provider decides on the default and we propagate that via late-initialization. If it's required and none of them set it, well, we can't know what they want and just show the provider's error saying that this is required. 

There is another case where resource class author may want to provide a default but also allow claimer to override it, like the `storageGB` example of yours, it's fair to support this as Crossplane. However, I think we need more discussion on how to achieve that because we have only 2 opinions `claimer's setting` and `resclass' setting` but this is a third one and IMO it deserves another way of representation. Maybe resource class could have `specTemplate.storageGB.override` as strong opinion and `specTemplate.storageGB.default` as default value.

Reply:
https://github.com/crossplaneio/crossplane/issues/91 is pretty close to what we're discussing here. I've linked this comment thread there. Let's close this and move further discussion there.